doc/src/atomvm-tooling.md

```diff
@@ -0,0 +1,239 @@
+<!--
+ Copyright 2023 Fred Dushin <fred@dushin.net>
+ SPDX-License-Identifier: Apache-2.0 OR LGPL-2.1-or-later
+-->
+
+# AtomVM Tooling
+
+AtomVM provides an implementation of the Erlang virtual machine, and as such it provides runtime support for applications targeted for the platform.
+
+However, developers will typically make use of downstream tooling that simplifies the development and provisioning of applications onto devices that are running the on the virtual machine.
+
+This chapter presents an overview of these tools and how they can be used to make you more productive as an AtomVM developer.
+
+
+## `atomvm_rebar3_plugin`
+
+The [`atomvm_rebar3_plugin`](https://github.com/atomvm/atomvm_rebar3_plugin) is a [rebar3](https://rebar3.org) plugin that can be used to create and flash Erlang applications that run over AtomVM.  Using this plugin greatly simplies the process of building Erlang applications that run over AtomVM, and is strongly encouraged for new users.
+
+### Prerequisites
+
+To use the [`atomvm_rebar3_plugin`](https://github.com/atomvm/atomvm_rebar3_plugin), you will need the following software on your development machine:
+
+* A suitable version of the [Erlang/OTP](https://www.erlang.org) distribution.  See the [Release Notes](release-notes.md) for information about supported Erlang/OTP versions.
+* A recent version of the [rebar3](https://rebar3.org) command-line tool.
+* (optional) The [git](https://git-scm.com) command line tool, to follow examples in this chapter.
```

Comment:
At this point should git be a requirement full stop? To use the plugin users should also point to the git repository in their own applications, at least until there is a CI workflow that publishes the plugin on `hex`.

Reply:
I dunno, do they need git installed?  The plugin is in hex, and even with rebar3 and git repos, I am not sure if that is not using https or ssh (?), as opposed to the `git` command.  I mean, I agree that everyone is going to have it, but unsure if if is needed.

Reply:
That is irrelevant now. Having it optional makes sense. We can work on a workflow later that can automatically publish releases to hex when a new release tag is made on GitHub, but as recent experience has shown it may be better to handle hex publishing manually until CI testing is more thoroughly tested.